xdsl/backend/riscv/register_allocation.py

```diff
@@ -49,6 +49,32 @@ def allocate(self, reg: SSAValue) -> bool:
 
         return False
 
+    def process_operation(self, op: Operation) -> None:
+        match op:
+            case riscv_scf.ForOp():
+                self.allocate_for_loop(op)
+            case RISCVOp():
+                self.process_riscv_op(op)
+            case _:
+                return
+
+    @abc.abstractmethod
+    def process_riscv_op(self, op: RISCVOp) -> None:
```

Comment:
I would appreciate some comments on this, e.g. the assumption that this is called from a reverse walk, etc. It would make it much clearer for readers to understand what each implementation of this actually does.